src/inmanta/data/__init__.py

```diff
@@ -2644,26 +2644,35 @@ async def get_latest_version(cls, environment: uuid.UUID, resource_id: m.Resourc
             return resources[0]
 
     @classmethod
-    def _get_released_resources_partial_base_query(cls):
+    def _get_released_resources_base_query(
+        cls, select_clause: str, environment_filter: Tuple[QueryType, object], offset: int
+    ) -> Tuple[str, List[object]]:
         """A partial query describing the conditions for selecting the latest released resources,
         according to the model version number."""
-        return f"""
+        subquery_filter_statement, subquery_values = cls.get_composed_filter_with_query_types(
+            offset=offset, col_name_prefix=None, environment=environment_filter
+        )
+        return (
+            f"""
+            {select_clause}
             FROM (
                 SELECT DISTINCT ON (r.resource_id) r.resource_id, r.attributes, r.resource_version_id, r.resource_type,
                     r.agent, r.resource_id_value, r.environment,
                     (CASE WHEN
                             (SELECT r.model < MAX(public.configurationmodel.version)
                             FROM public.configurationmodel
                             WHERE public.configurationmodel.released=TRUE
-                            GROUP BY environment)
+                            AND {subquery_filter_statement})
                         THEN 'orphaned'
                     ELSE r.status::text END) as status
                 FROM {cls.table_name()} as r
                     JOIN public.configurationmodel as cm ON r.model=cm.version
```

Comment:
This I think also has to be constrained to a single environment otherwise it can join on the same version in another environment

Reply:
Nice catch :+1: it was indeed necessary

Reply:
this thing scares me...